src/Http/Middleware/SsoLogger.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Northwestern\SysDev\SOA\Http\Middleware;
+
+use http\ENV;
+use Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode as Middleware;
+use Auth;
+use App\Access;
+use Closure;
+use Illuminate\Support\Facades\Log;
+
+class SsoLogger extends Middleware
+{
+    protected $max_logs;
+
+    public function __construct() { $this->max_logs = ENV('SSO_LOG_MAX_ENTRIES',1000000); }
+
+    public function handle($request, Closure $next, ...$guards)
+    {
+        $controller = $request->route()->controller;
+
+        if (class_basename($controller) == "WebSSOController") {
+            // They're doing auth right now so we'll call it an anonymous request (no netid)
+            $this->log_request(true, $request);
+            return $next($request);
+        } else {
+            if (Auth()->user() == null) {
+                $this->log_request(true, $request);
+            } else {
+                $this->log_request(false, $request);
+            }
+        }
+
+
+        if (sizeof(Access::all()) > $this->max_logs) {
+            $this->remove_oldest_logs();
+        };
+
+        return $next($request);
+    }
+
+    private function log_request($anonymous, $request) {
+        $path = $request->path();
+        try {
+            $agent = $request->header('User-Agent');
+        }  catch (Exception $x) {
+            $agent = null;
+        }
+        $access = new Access(['path'=>$path,'agent'=>$agent]);
+        $time = Time();
+        if (!$anonymous) {
+            $netid = Auth::user()['netid'];
```

Comment:
This is an unsafe assumption to make in a package -- laravel-soa does not control the user model, so we don't necessarily know what the username attribute is. 

I'd be using this in an app where that is actually `username`, since I have multiple login methods :grin:

As far as agnostic ways to get the username go...anything object you can log in as implements the [`Authenticatable`](https://github.com/laravel/framework/blob/6.x/src/Illuminate/Contracts/Auth/Authenticatable.php#L19) interface. Sadly, that get identifier method gives back the `id` for the default Eloquent object. I'm not sure how useful that is.

I'm thinking you may need to add your own interface &  note it in the docs. People would have to add `implements SsoLoggable` to their `User` model and define a method that gives you the username. 

You can check to see if they've done so when you log and fall back on the `getAuthIdentifier()` method maybe? (id being better than nothing 🤷‍♂ )